server/ip-tools.ts

```diff
@@ -108,16 +108,20 @@ export const IPTools = new class {
 	ipToNumber(ip: string) {
```

Comment:
Should this return `number | null`? `null` is a better error state (easier for TypeScript to catch if you forgot to handle it).

Reply:
Yeah, probably